internal/provider/webhook_data_source.go

```diff
@@ -0,0 +1,197 @@
+package provider
+
+import (
+	"conductorone/internal/sdk"
+	"conductorone/internal/sdk/pkg/models/operations"
+	"conductorone/internal/sdk/pkg/models/shared"
+	"conductorone/internal/validators"
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+)
+
+func NewWebhookDataSource() datasource.DataSource {
+	return &WebhookDataSource{}
+}
+
+// WebhookResource defines the data source implementation.
+type WebhookDataSource struct {
+	client *sdk.ConductoroneAPI
+}
+
+// WebhookResourceModel describes the resource data model.
+type WebhookDataSourceModel struct {
+	CreatedAt   types.String `tfsdk:"created_at"`
+	DeletedAt   types.String `tfsdk:"deleted_at"`
+	Description types.String `tfsdk:"description"`
+	DisplayName types.String `tfsdk:"display_name"`
+	ID          types.String `tfsdk:"id"`
+	UpdatedAt   types.String `tfsdk:"updated_at"`
+	URL         types.String `tfsdk:"url"`
+}
+
+func (r *WebhookDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_webhook"
+}
+
+func (r *WebhookDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Webhook Data Source",
+
+		Attributes: map[string]schema.Attribute{
+			"created_at": schema.StringAttribute{
+				Computed: true,
+				Validators: []validator.String{
+					validators.IsRFC3339(),
+				},
+			},
+			"deleted_at": schema.StringAttribute{
+				Computed: true,
+				Validators: []validator.String{
+					validators.IsRFC3339(),
+				},
+			},
+			"description": schema.StringAttribute{
+				Computed:    true,
+				Description: `The description field.`,
+			},
+			"display_name": schema.StringAttribute{
+				Computed:    true,
+				Optional:    true,
+				Description: `The displayName field.`,
+			},
+			"id": schema.StringAttribute{
+				Computed:    true,
+				Optional:    true,
+				Description: `The id field.`,
+			},
+			"updated_at": schema.StringAttribute{
+				Computed: true,
+				Validators: []validator.String{
+					validators.IsRFC3339(),
+				},
+			},
+			"url": schema.StringAttribute{
+				Computed:    true,
+				Description: `The url field.`,
+			},
+		},
+	}
+}
+
+func (r *WebhookDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
```

Comment:
Is this auto generated? I wonder if its worth asking SE to throw an error when the provider is not configured.

Reply:
Yep, that part is auto generated